source/dhtslib/vcf/header.d

```diff
@@ -1,8 +1,260 @@
 module dhtslib.vcf.header;
 
+import std.datetime;
 import std.string: fromStringz, toStringz;
+import std.format: format;
+import std.traits : isArray, isIntegral, isSomeString;
+import std.conv: to, ConvException;
+import std.algorithm : map;
+import std.array : array;
+import std.utf : toUTFz;
 
+import dhtslib.vcf;
 import htslib.vcf;
+import htslib.hts_log;
+
+
+/// Struct for easy setting and getting of hrec values for VCFheader
+struct HeaderRecord

```

Comment:
I made this comment elsewhere, but is it possible to write these structs so that they introspect the existing htslib-formatted records in memory (perhaps needing to unpack them, on as needed basis just as htslib does), so that we avoid data copies?

My major concern is that, while very ergonomic, copying headers, copying records, copying info fields into D structs for every row of a large file wil incur massive speed penalty -- perhaps I'm wrong and IO is the limiting factor, but I think you should run some benchmarks on this branch before we accept the PR (or before we try to rewrite it which I acknolwedge would be a big effort)

Reply:
Now, header records are the least offensive since there are a small-ish fixed number per file. My major concern is with row records, and field records (info, format)

Reply:
Can you expoud upon how `HeaderRecord` is used?

Reply:
> Now, header records are the least offensive since there are a small-ish fixed number per file. My major concern is with row records, and field records (info, format)

I agree I wasn't bothered by this behavior for the header records and this is why it is the most ergonomic. I think the `InfoField` and `FormatField` structs should probably be changed to avoid the copy.

> Can you expoud upon how HeaderRecord is used?

I thought is would be a nice feature to have as for instance in `recontig` we do a lot of header manipulation during conversion. This involves pulling different parts of the existing header and altering it. I wanted a way I could piece by piece build a header record without resorting to string manipulation. Of course, you should still be able to just specify a header line as a string, but I would like the ability to build it in a more structured way when needed.